finality-aleph/src/sync/service.rs

```diff
@@ -119,7 +130,7 @@ where
                 justifications_from_user,
                 additional_justifications_from_user,
                 block_requests_from_user,
-                metrics,
+                metrics: (metrics, top_block_metrics),
```

Comment:
Just make them one  struct, i.e. add the top block metrics to the previous struct. If you want the logic to be more separated then make it have two substructs within the `Prometheus` variant.

Also, since I'm already commenting despite not being a reviewer, maybe consider keeping the `noop` constructor? It's somewhat cleaner to not depend on the (imo internal implementation) detail that noop is implemented as an enum variant.

Reply:
Sure, combined them. The substruct solution would require having methods in outer struct, or else pattern matching, so chosen simplest option. Bonus: I've just learned that in substrate, sometimes the following pattern is used:
```
pub fn report(&self, do_this: impl FnOnce(&Metrics)) {
    if let Some(metrics) = self.0.as_ref() {
	do_this(metrics);
    }
}
```
But calling this later `self.metrics.report(|metrics| metrics.some_fn())` looks perhaps more complicated than it should.